V3 thermal node

zrezke opened the pull request:
- Moved thermal to it's own node.
- Should be expanded later on to include runtime controls for tiny1c.
- And also it might be worth to create a ThermalFrame dtype with methods to convert between temperature units and get temp at X,Y.
- Right now you can get a FP16 and/or YUV422i ImgFrame from the temperature and color outputs respectively.

moratom commented on examples/python/RVC2/Thermal/thermal.py:
Let's change to camelCase to keep examples consistent.

zrezke replied:
Oh, sorry, I always forget we use camelCase here...

moratom reviewed (commented):
Thanks! I'd say we first implement the ThermalConfig on device side so it's correctly used and we can merge the PR.

moratom commented on include/depthai/pipeline/datatype/ThermalConfig.hpp:
```diff
@@ -0,0 +1,138 @@
+/// Orientation of the image.
+enum ThermalImageOrientation { Normal, Mirror, Flip, MirrorFlip };
+
+class ThermalImageParams : public Buffer {
+   public:
+    /// 0-3. Time noise filter level. Filters out the noise that appears over time.
+    std::optional<uint8_t> timeNoiseFilterLevel;
+
+    /// 0-3. Spatial noise filter level.
+    std::optional<uint8_t> spatialNoiseFilterLevel;
+
+    /// 0-4 Digital etail enhance level.
+    std::optional<uint8_t> digitalDetailEnhanceLevel;
+
+    /// Image brightness level, 0-255.
+    std::optional<uint8_t> brightnessLevel;
+
+    /// Image contrast level, 0-255.
+    std::optional<uint8_t> contrastLevel;
+
+    /// Orientation of the image. Computed on the sensor.
+    std::optional<ThermalImageOrientation> orientation;
+    DEPTHAI_SERIALIZE(
+        ThermalImageParams, timeNoiseFilterLevel, spatialNoiseFilterLevel, digitalDetailEnhanceLevel, brightnessLevel, contrastLevel, orientation);
+};
```
Doesn't need to inherit from Buffer

zrezke commented:
This is ready now. @moratom

zrezke commented:
@moratom , I added the crosshair example and fixed the autoFFC not working + in theory scaled up the displayed images, but it seems like the opencv resizeWindow function doesn't work like it used to, reported the issue here: https://github.com/opencv/opencv/issues/26180

So for opencv-python<4.10.0.82 the image resize will work, for newer versions not
